### DISCUSSION TITLE
findFirst where relation count is greater than zero

### DISCUSSION THREAD

**Discussion by pishguy** [prisma/prisma#21553] [Q&A]
In this simple query, how can I add gt to select all subCategories when count of menu is greater than zero?

### How to reproduce

```dart
return await this.prisma.category.findFirst({
  where: {
    id: categoryId,
  },
  include: {
    subCategories: {
      include: {
        menu: true,
      },
    },
  },
});
```

This query returns all menus, even menus that have a count of zero.

**Comment by ludralph** [comment]
Hi @pishguy 👋 

Thank you for raising this question.

Prisma currently does not support filtering on relation counts directly in the query. This means you cannot use the `gt` operator to filter `subcategories` based on the count of related `menu` items directly in the `findFirst` or `findMany` query. However, you can fetch the data first and then filter the results in your application code.

There is an open [feature request](https://github.com/prisma/prisma/issues/8935) for supporting filtering by relation counts in Prisma.
You can add your use case and leave a 👍 on the open issue to help our engineers prioritize it.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by bvanderdrift** [comment]
You can use the edge-case results of `every` and `some` to trick the result, specifically for `0` cases.

For example to get all categories & subcategories with only 1 or more menu's (I'm not sure I understand you use case completely).

```js
await prisma. category.findFirst({
    where: {
      id: categoryId,
    },
    include: {
      subCategories: {
        include: {
          menu: true,
        },
        where: {
          NOT: {
            menu: {
              // This will match if there are no menu's
              every: {
                id: 'some impossible id'
              }
            }
          }
        }
      },
    },
  });
```

This is similar to

```js
[].every(element => element === 3); // true
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
